File: mcp-server/packages/context-extractor-csharp/tests/Examples/NotificationService.cs
namespace ContextExtractor.Tests.Examples;

/// <summary>
/// Service for sending notifications.
/// </summary>
public class NotificationService
{
    public Logger Logger { get; }

    public NotificationService(Logger logger)
    {
        Logger = logger;
    }

    public bool SendEmail(string address, string subject, string content)
    {
        return true;
    }

    public bool SendSms(string number, string content)
    {
        return true;
    }

    public bool SendPush(string deviceToken, string title, string body)
    {
        return true;
    }
}

File: mcp-server/packages/context-extractor-java/src/main/java/io/contextextractor/examples/NotificationService.java
package io.contextextractor.examples;

/**
 * Service for sending notifications.
 */
public class NotificationService {
    private final Logger logger;
    
    public NotificationService(Logger logger) {
        this.logger = logger;
    }
    
    public boolean sendEmail(String address, String subject, String content) {
        return true;
    }
    
    public boolean sendSms(String number, String content) {
        return true;
    }
    
    public boolean sendPush(String deviceToken, String title, String body) {
        return true;
    }
    
    public Logger getLogger() {
        return logger;
    }
}
